Lib/test/test_interpreters/test_api.py

```diff
@@ -1414,6 +1414,41 @@ def test_call_invalid(self):
             with self.assertRaises(interpreters.NotShareableError):
                 interp.call(func, op, 'eggs!')
 
+    def test_callable_requires_frame(self):
+        # There are various functions tha require a current frame.
+        interp = interpreters.create()
+        for call, expected in [
+            ((eval, '[1, 2, 3]'),
+                [1, 2, 3]),
+            ((eval, 'sum([1, 2, 3])'),
+                6),
+            ((exec, '...'),
+                None),
+        ]:
+            with self.subTest(str(call)):
+                res = interp.call(*call)
+                self.assertEqual(res, expected)
+
+        notshareable = [
+            globals,
+            locals,
+            vars,
+        ]
```

Comment:
If "notshareable" here means that the result of the functions cannot be pickled, could `NotShareableError` differentiate the round-trip in the future?  Also, the `is_shareable()` module method returns `False` for both `globals` and `dir`, which might be confusing at first.

Reply:
fixed